concourse/scripts/bosh_create_and_upload_release.rb

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env ruby
+require 'cli'
+
+def usage
+  puts "Usage: bosh_create_and_upload_release.rb <release_name> <release_version> <release_dir>"
+  exit 101
+end
+
+def check_release(name, version)
+  bosh_cli = Bosh::Cli::Command::Base.new.director
+
+  this_release = bosh_cli.list_releases.select{ |r| r["name"] == name }.first
+  unless this_release.nil? or this_release.empty?
+    this_release_version = this_release["release_versions"].select { |rv|
+      rv["version"] == version
+    }.first
+  end
+  this_release_version
+end
+
+def create_release(name, version, dir)
+  cr_cmd = Bosh::Cli::Command::Release::CreateRelease.new
+  cr_cmd.add_option(:name, name)
+  cr_cmd.add_option(:version, version)
+  cr_cmd.add_option(:dir, dir)
+  cr_cmd.add_option(:with_tarball, true)
+  Bosh::Cli::Config.output = STDOUT
+
+  puts "Create release #{name} version #{version}"
+
+  begin
+    cr_cmd.create
+  rescue Bosh::Cli::CliError => e
+    if e.message.include?("Release version already exists")
+      puts "Release version already exists locally"
+    else
+      raise e
+    end
+  end
+
+end
+
+def upload_release(name, version, dir)
+  release_filename = "dev_releases/#{name}/#{name}-#{version}.tgz"
```

Comment:
/cc @Jonty 
> dev_releases is only mentioned here - should it be hardcoded?

`Bosh::Cli::Command::Release::CreateRelease` creates a "development release" unless it is specified as "final". Dev releases are always drop in that directory to be uploaded manually. 

Final releases are not drop in a local directory, they must be uploaded to the blobstore configured in `config/final.yml`